tests/ParserHelperTests.cpp

```diff
@@ -98,6 +98,38 @@ TEST(ParserHelper_Tests, IgnoreStringIgnoresWholeQuoation)
 	ASSERT_EQ(" text", std::string{buffer});
 }
 
+
+TEST(ParserHelper_Tests, stringToIntegerLogsNotFullyMatchingInput)
+{
+	std::string str{R"(345 foo)"};
+
+	const std::stringstream log;
+	auto* const stdOutBuf = std::cout.rdbuf();
+	std::cout.rdbuf(log.rdbuf());
+
+	const auto theInteger = commonItems::stringToInteger<int>(str);
+
+	std::cout.rdbuf(stdOutBuf);
+
+	ASSERT_EQ(" [WARNING] string to integer: invalid argument! 345 foo\n", log.str());
```

Comment:
Yeah, I'd skip warnings. I regularly use stoi("345 - ProvinceName.txt") to get 345, no fuss about it.

Reply:
I've added a `bool skipPartialMatchWarning` parameter which defaults to false.